presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplitManager.java

```diff
@@ -140,11 +140,13 @@ public ConnectorSplitSource getPartitionSplits(ConnectorSession session, Connect
         private final TupleDomain<RaptorColumnHandle> effectivePredicate;
         private final CloseableIterator<ShardNodes> iterator;
 
+        private CompletableFuture<List<ConnectorSplit>> future;
```

Comment:
Is this class meant to be thread-safe? This non-final field is initialized and accessed without any protection